packages/editor/src/components/theme-support-check/index.js

```diff
@@ -1,20 +1,22 @@
 /**
  * WordPress dependencies
  */
-import { withSelect } from '@wordpress/data';
+import { useSelect } from '@wordpress/data';
 import { store as coreStore } from '@wordpress/core-data';
 
 /**
  * Internal dependencies
  */
 import { store as editorStore } from '../../store';
 
-export function ThemeSupportCheck( {
-	themeSupports,
-	children,
-	postType,
-	supportKeys,
-} ) {
+export default function ThemeSupportCheck( { children, supportKeys } ) {
+	const { postType, themeSupports } = useSelect( ( select ) => {
+		return {
+			postType: select( editorStore ).getEditedPostAttribute( 'type' ),
+			themeSupports: select( coreStore ).getThemeSupports(),
+		};
+	}, [] );
```

Comment:
I wasn't immediately sure if a single `useSelect()` or multiple ones make more sense, due to the fact that we're subscribing to multiple stores.

Fine to keep as-is, since it was a single call before.

Reply:
> Fine to keep as-is, since it was a single call before.

That and if you scan the core, there are a lot of places where `core` and `editor` selectors are used together.